jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamProjectWizard.kt

```diff
@@ -68,11 +71,28 @@ abstract class SamProjectTemplate {
 
     protected fun addSourceRoots(project: Project, modifiableModel: ModifiableRootModel, projectRoot: VirtualFile) {
         val template = SamCommon.getTemplateFromDirectory(projectRoot) ?: return
-        val codeUris = SamCommon.getCodeUrisFromTemplate(project, template)
-        modifiableModel.contentEntries.forEach { contentEntry ->
-            if (contentEntry.file == projectRoot) {
-                codeUris.forEach { contentEntry.addSourceFolder(it, false) }
+        val templatePath = Paths.get(template.parent.path)
+        runInEdt {
+            val functions = SamTemplateUtils.findFunctionsFromTemplate(project, template)
+            val functionLocations = functions.map {
+                val codeLocation = SamTemplateUtils.getCodeLocation(template.toNioPath().toAbsolutePath(), it.logicalName)
+                if(codeLocation.startsWith("./")){
+                    templatePath.resolve(codeLocation.substring(2))
+                } else {
+                    templatePath.resolve(codeLocation)
+                }
```

Comment:
I dont think you need to check "./", the path API should always relativize it if its not "/"

I would do `templatePath.parent.resolve(codeLocation)`

Reply:
Made the change